api/client.go

```diff
@@ -1249,6 +1249,83 @@ START:
 	return result, nil
 }
 
+// requestWithContext avoids the use of the go-retryable library found in RawRequestWithContext and is
+// useful when making requests where the req or resp body is likely larger than available memory
+func (c *Client) requestWithContext(ctx context.Context, r *Request) (*Response, error) {
```

Comment:
Should also note in godoc that we ignore *all* config settings, including ones that aren't retryable-specific, i.e.. HttpClient, Timeout, OutputCurlString, Logger.

Reply:
godoc updated to be more clear as to what this method should be used for and what config options apply to it. I've also made it a bit more clear in the code to show that we use the `HttpClient` and fail-fast for `OutputCurlString`